### ISSUE TITLE
When running a cluster with 1 node pool, create new node pool before destroying old one

### ISSUE THREAD

**Issue by forestoden** [terraform-google-modules/terraform-google-kubernetes-engine#1289]
### TL;DR

When running a cluster with one node pool and making a change that forces a recreation, the order of operations is to delete the node pool then create the new one. It would be much more ideal to create the new node pool first so that we don't end up a situation where the cluster has no node pools

### Expected behavior

When updating a cluster with one node pool, the new node pool is created first, then the old one is destroyed. 

### Observed behavior

When updating a cluster with one node pool, the new node pool is created first, then the old one is destroyed. 

### Terraform Configuration

```hcl
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  version = "~> 21.0"

  project_id         = var.project_id
  name               = "name"
  region             = var.region


  remove_default_node_pool        = true
  horizontal_pod_autoscaling      = true
  enable_vertical_pod_autoscaling = true
  network_policy                  = true
  http_load_balancing             = true
  enable_private_nodes            = true
  default_max_pods_per_node       = 32


  node_pools = [
    {
      name               = "nodepool" # change name or something to force a recreation
      machine_type       = "e2-standard-4"
      min_count          = 1
      max_count          = 3
      disk_size_gb       = 50
      disk_type          = "pd-standard"
      image_type         = "COS_CONTAINERD"
      initial_node_count = 1
    },
  ]
}
```

### Terraform Version

```sh
v1.2.1
```

### Additional information

I'm not even sure if this possible with Terraform but wanted to put it out there.

**Comment by bharathkkb** [comment]
Hi @forestoden PTAL at our update-variant modules which creates new nodepools before deletion. Example: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/beta-public-cluster-update-variant#node-pool-update-variant
